fish-rust/src/ast.rs

```diff
@@ -160,6 +161,18 @@ trait NodeMut: Node + AcceptorMut + ConcreteNodeMut {
     fn as_node(&self) -> &dyn Node;
 }
 
+macro_rules! fn_return_option_none {
+    ($($name:ident),*) => {
+        $(
+            paste!{
+                fn [<as_ $name>](&self) -> Option<&[<$name:camel>]> {
```

Comment:
interesting, I didn't know this was possible without defining a proc macro

Reply:
To be clear `paste!{}` *is* a proc macro, it's just being wrapped up in another declarative macro.